src/utils/discord_util.py

```diff
@@ -1,13 +1,47 @@
 """
     Discord Utility module
 """
-
+from discord import Intents, Client
 from .command_interface import CommandInterface
 
+intents = Intents.default()
+intents.message_content = True
+client = Client(intents=intents)
+
+
 class DiscordUtil(CommandInterface):
     """
-        A class for performing all logic to interact with Discord as a bot
+    A class for performing all logic to interact with Discord as a bot
     """
 
     def __init__(self, discord_auth_token):
+        self.token = discord_auth_token
+
+        self.run()
         pass
+
+    @client.event
+    async def on_ready(self):
+        """
+        Send a notification when the bot is connected and ready to recieve events
+        """
+        print(f"Logged in as {client.user}")
+
+    @client.event
+    async def on_message(self, message):
+        """
+        Handle incoming events
+        """
+        if message.author == client.user:
+            return
+        channel = message.channel
```

Comment:
This is good, but we only want the bot to listen for commands on the `#leetcode` channel. The `#leetcode` channel ID is `1072303681501941791`